samples/go/csv/csv-export/exporter.go

```diff
@@ -46,8 +47,16 @@ func main() {
 	err = d.Try(func() {
 		defer profile.MaybeStartProfile().Stop()
 
-		nomsList, structDesc := csv.ValueToListAndElemDesc(ds.HeadValue(), ds.Database())
-		csv.Write(nomsList, structDesc, comma, os.Stdout)
+		hv := ds.HeadValue()
+		if l, ok := hv.(types.List); ok {
+			structDesc := csv.GetListElemDesc(l, ds.Database())
+			csv.WriteList(l, structDesc, comma, os.Stdout)
+		} else if m, ok := hv.(types.Map); ok {
+			structDesc := csv.GetMapElemDesc(m, ds.Database())
+			csv.WriteMap(m, structDesc, comma, os.Stdout)
+		} else {
+			panic(fmt.Sprintf("Expected ListKind or MapKind, found %s", types.KindToString[hv.Type().Kind()]))
+		}
 	})
```

Comment:
What happens here if the user is exporting to a brand new dataset? Seems like that's not possible.